csrc/executor.cpp

```diff
@@ -1868,14 +1999,8 @@ std::vector<at::Tensor> FusionExecutor::runFusion(
     }
   }
 
-  std::vector<std::vector<std::byte>> arg_buffers;
-  {
-    FUSER_PERF_SCOPE("ExecutorRunFusion::GetArgsBuffers");
-    arg_buffers.reserve(kernel()->parameters().size());
-    for (auto v : kernel()->parameters()) {
-      arg_buffers.emplace_back(
-          getKernelArgument(expr_eval, v, kernel()->indexType()));
-    }
+  if (executor_entry->args.empty()) {
```

Comment:
@rdspring1 note that I didn't change the deserialization path; instead I modified this line to just ask if the relevant field was initialized (before, when hitting the issue we discussed, I had kept track of whether `initializeExecutorEntry` was called and use that as the conditional).

Something deep in the `ExpressionEvaluator` was not initialized. It was probably that, at the time of `::deserialize`, we don't have access to the `KernelArgumentHolder` and so I couldn't `ExpressionEvaluator::bind` the inputs. Or maybe it was the `FusionExecutor::kernel()` itself wasn't initialized?

I didn't spend too much time trying to figure it out; it doesn't seem like a terrible thing that the first launch after deserialization is slower than subsequent launches. This is the same for non-deserialization cases anyway.

Of potential note is that even in the normal/non-deserialized case, the `computeArgs` call needs to be this late. Something is getting initialized before this that is critical for us to identify the argument information.

Reply:
Serialization associates a metadata `KernelArgumentHolder` for each `FusionKernelRuntime`. We don't have full arguments available during deserialization.

Reply:
IIUC, you could fill the `ExpressionEvaluator` with the metadata `KernelArgumentHolder` and run `computeArgs`.

Reply:
I think I'd need full args. I tried to bind kernel()->inputs(), but that died somewhere deep with an assert that something was a std::monostate instead of a real object.